samples/javascript_nodejs/52.teams-messaging-extensions-search-auth-config/bots/teamsMessagingExtensionsSearchAuthConfigBot.js

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler, CardFactory, ActionTypes } = require('botbuilder');
+const axios = require('axios');
+const querystring = require('querystring');
+const { SimpleGraphClient } = require('..\\simpleGraphClient.js');
+
+// User Configuration property name
+const USER_CONFIGURATION = 'userConfigurationProperty';
+
+class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
+    /**
+     *
+     * @param {UserState} User state to persist configuration settings
+     */
+    constructor(userState) {
+        super();
+        // Creates a new user property accessor.
+        // See https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors.
+        this.userConfigurationProperty = userState.createProperty(USER_CONFIGURATION);
+        this.connectionName = process.env.ConnectionName;
+        this.userState = userState;
+    }
+
+    async handleTeamsMessagingExtensionConfigurationQuerySettingUrl(context, query) {
+        // The user has requested the Messaging Extension Configuration page settings url.
+        const userSettings = await this.userConfigurationProperty.get(context, '');
+        let escapedSettings = '';
+        if (userSettings) {
+            escapedSettings = querystring.escape(userSettings);
+        }
+
+        return {
+            composeExtension: {
+                type: 'config',
+                suggestedActions: {
+                    actions: [
+                        {
+                            type: ActionTypes.OpenUrl,
+                            value: process.env.SiteUrl + '/public/searchSettings.html?settings=' + escapedSettings
```

Comment:
better to use template literal (aka back tick string) rather than adding strings

Reply:
fixed